lab_5_scrapper/scrapper.py

```diff
@@ -90,25 +189,40 @@ def __init__(self, config: Config) -> None:
         """
         Initializes an instance of the Crawler class
         """
-        pass
+        self.seed_urls = config.get_seed_urls()
+        self.config = config
+        self.urls = []
 
     def _extract_url(self, article_bs: BeautifulSoup) -> str:
         """
         Finds and retrieves URL from HTML
         """
-        pass
+        return article_bs.get('href')
 
     def find_articles(self) -> None:
         """
         Finds articles
         """
-        pass
+        for seed_url in self.seed_urls:
+            response = make_request(seed_url, self.config)
+            if response.status_code != 200:
+                continue
+            main_bs = BeautifulSoup(response.text, 'lxml')
+            all_links_bs = main_bs.find_all('a')
+            for link_bs in all_links_bs:
+                href = self._extract_url(link_bs)
+                if href is None:
+                    print(link_bs)
+                    continue
+                if (href.startswith('/news') or href.startswith('/gov') or href.startswith(
+                        '/society') or href.startswith('/business')) and href.count('/') == 3 and 'comment' not in href:
+                    self.urls.append("https://chelny-biz.ru" + link_bs['href'])
```

Comment:
why using `links['href']` instead of `href`?

Reply:
actually I don't know... I've changed this part